common/src/common/misc/SpecialFunctions.cpp

```diff
@@ -107,6 +108,21 @@ void SpecialFunctions::mapStart(Functional* obj, const j3x::Obj& data, Character
     // empty
 }
 
+void SpecialFunctions::changeMapParams(Functional* obj, const j3x::Obj& data, Character* user)
+{
+    LOG.info("[SpecialFunction] Change map params.");
+
+    auto& list = j3x::getObj<j3x::List>(data);
+    auto& shader = j3x::getObj<std::string>(list, 0);
+    auto& bg_color = j3x::getObj<int>(list, 1);
+    auto& light_color = j3x::getObj<int>(list, 2);
+    j3x::Parameters new_parameters = {{"shader",           shader},
+                                      {"background_color", bg_color},
+                                      {"lighting_color",   light_color}};
+    framework_->getMap()->setParams(new_parameters);
+    framework_->refreshColors();
```

Comment:
I suppose this should be added to Journal when invoked? What do you think @majewskimaciej ?

Reply:
I think so. It would look weird if returning to a room with different parameters through time reversal didn't change them back.

Reply:
done